src/cozmo/pets.py

```diff
@@ -0,0 +1,250 @@
+:class:`cozmo.world.World` object, so events for all known pets can be
+observed by adding handlers there.
+'''
+
+# __all__ should order by constants, event classes, other classes, functions.
+__all__ = ['PET_VISIBILITY_TIMEOUT', 'PET_TYPE_UNKOWN', 'PET_TYPE_CAT', 'PET_TYPE_DOG',
+           'EvtPetAppeared', 'EvtPetDisappeared', 'EvtPetObserved',
+           'Pet']
+
+
+import math
+import time
+
+from . import logger
+
+from . import event
+from . import objects
+from . import util
+
+from ._clad import _clad_to_game_anki
+
+
+#: Length of time to go without receiving an observed event before
+#: assuming that Cozmo can no longer see a pet.
+PET_VISIBILITY_TIMEOUT = objects.OBJECT_VISIBILITY_TIMEOUT
+
+# Pet types that Cozmo can distinguish
+#: Pet Type reported by Cozmo when unsure of type of pet
+PET_TYPE_UNKOWN = "Unknown"
+#: Pet Type reported by Cozmo when he thinks it's a cat
+PET_TYPE_CAT = "Cat"
+#: Pet Type reported by Cozmo when he thinks it's a dog
+PET_TYPE_DOG = "Dog"
+
+
+class EvtPetObserved(event.Event):
+    '''Triggered whenever a pet is visually identified by the robot.
+
+    A stream of these events are produced while a pet is visible to the robot.
+    Each event has an updated image_box field.
+
+    See EvtPetAppeared if you only want to know when a pet first
+    becomes visible.
+    '''
+    pet = 'The Pet instance that was observed'
+    updated = 'A set of field names that have changed'
+    image_box = 'A comzo.util.ImageBox defining where the pet is within Cozmo\'s camera view'
+
+
+class EvtPetAppeared(event.Event):
+    '''Triggered whenever an object is first visually identified by a robot.
+
+    This differs from Evt{etObserved in that it's only triggered when
+    a pet initially becomes visible.  If it disappears for more than
+    PET_VISIBILITY_TIMEOUT (0.2) seconds and then is seen again, a
+    EvtPetDisappeared will be dispatched, followed by another
+    EvtPetAppeared event.
+
+    For continuous tracking information about a visible pet, see
+    EvtPetObserved.
+    '''
+    pet = 'The Pet instance that was observed'
+    updated = 'A set of field names that have changed'
+    image_box = 'A comzo.util.ImageBox defining where the pet is within Cozmo\'s camera view'
+
+
+class EvtPetDisappeared(event.Event):
+    '''Triggered whenever a pet that was previously being observed is no longer visible.'''
+    pet = 'The Pet instance that is no longer being observed'
+
+
+def _clad_pet_type_to_pet_type(clad_pet_type):
+    if clad_pet_type == _clad_to_game_anki.Vision.PetType.Unknown:
+        return PET_TYPE_UNKOWN
+    elif clad_pet_type == _clad_to_game_anki.Vision.PetType.Cat:
+        return PET_TYPE_CAT
+    elif clad_pet_type == _clad_to_game_anki.Vision.PetType.Dog:
+        return PET_TYPE_DOG
+    else:
+        raise ValueError("Unexpected pet type %s" % clad_pet_type)
+
+
+class Pet(event.Dispatcher):
+    '''A single pet (face) that Cozmo has detected.
```

Comment:
Maybe remove "(face)"? Most or all other references to pet just say "pet", not "pet's face" or similar.